### PR TITLE
Grammar experiments

### PR DESCRIPTION
**Pull request by favonia**

### CODE REVIEW

**Comment by favonia** [comment]
@jonsterling @cangiuli @ivoysey If we want to have n-ary tuples and make the lambda symbol optional (that is, `x => x` for the identity function), then we have to decide the meaning of...

```
[ a => b ]
```

Is it a system with one branch, or a unary tuple with a function? Also, why would not a user curse us? Please make an informed wish. :smiley_cat:

**Comment by favonia** [comment]
What should we do about `λ elim`?

File: src/lib/Grammar.mly [hunk 1]
```diff
@@ -66,30 +86,41 @@ sign:
   | d = decl; s = sign
     { d :: s }
 
-plain_atomic_cof_except_atomic_term:
-  | BOUNDARY t = term
+plain_atomic_in_cof_except_term:
+  | BOUNDARY t = atomic_term
```
Comment (favonia) [inline src/lib/Grammar.mly:90]: I decided to weaken the expressiveness in this particular case... unless people make another wish at #69.

**Comment by jonsterling** [comment]
Hmm, what if we got rid of the non-lambda `elim`? Like, if you want to write `elim n @ mot ...]` you must instead write let `f : (n : nat) -> mot = elim [...] in f n`?

**Comment by favonia** [comment]
I see your points... hmm... not sure.

However, I do not think I have to change any grammar for this wish---you are asking "what if I don't want to write a program in a particular way?" Just don't write it! Anyways, does that mean we should merge this PR first?

**Comment by jonsterling** [comment]
@favonia The reason to get rid of that production is not to prevent the user from writing some code, but to enable us to change the grammar to be more immortal: to be able to write `elim` instead of `\elim`, as I have done in my commit above.

**Comment by favonia** [comment]
> @favonia The reason to get rid of that production is not to prevent the user from writing some code, but to enable us to change the grammar to be more immortal: to be able to write `elim` instead of `\elim`, as I have done in my commit above.

I should have said this instead: as a genie, I am not interested in things that are humanly possible. :stuck_out_tongue_closed_eyes:

**Pull request merged**